data/src/test/java/com/microsoft/azure/kusto/data/ResultSetTest.java

```diff
@@ -0,0 +1,93 @@
+package com.microsoft.azure.kusto.data;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class ResultSetTest {
+    @Test
```

Comment:
Ideally should be a separate test for each unit being tested.

If that would take too long, might be worth at least checking each return type with 1 unit test (e.g. input string value, input int value and input column name all in 1 unit test).

Reply:
as talked - i dont think i should do this here as its too much code